File: frontend/src/services/gmail.ts
import { apiRequest } from "./api";

export type GmailAuthUrlData = {
  auth_url: string;
};

export type GmailStatusData = {
  connected: boolean;
  gmail_account_hint: string | null;
  last_checked_at: string | null;
};

export type GmailPollData = {
  polled?: boolean;
  processed_messages: number;
  matched_notifications?: number;
};

export async function getGmailAuthUrl(): Promise<GmailAuthUrlData> {
  return apiRequest<GmailAuthUrlData>("/api/v1/gmail/auth-url");
}

export async function getGmailStatus(): Promise<GmailStatusData> {
  return apiRequest<GmailStatusData>("/api/v1/gmail/status");
}

export async function pollGmailNow(): Promise<GmailPollData> {
  return apiRequest<GmailPollData>("/api/v1/gmail/poll", {
    method: "POST",
  });
}

File: backend/app/schemas/gmail.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel


class GmailAuthUrlData(BaseModel):
    auth_url: str


class GmailConnectData(BaseModel):
    connected: bool
    has_refresh_token: bool
    gmail_account_hint: str | None = None
    email: str | None = None


class GmailStatusData(BaseModel):
    connected: bool
    gmail_account_hint: str | None = None
    email: str | None = None
    last_checked_at: datetime | None = None


class GmailPollData(BaseModel):
    polled: bool
    processed_messages: int
    matched_notifications: int
